src/Classifier.php

```diff
@@ -105,7 +117,7 @@ public function find(): iterable
                 }
             }
 
-            if (($this->parent !== null) && !is_subclass_of($className, $this->parent)) {
+            if (($this->targetClass !== null) && !is_subclass_of($className, $this->targetClass)) {
```

Comment:
`targetClass` looks like a class that you exactly want to find, but actually it finds also inheriting classes